src/analysis/normed_space/star/continuous_functional_calculus.lean

```diff
@@ -0,0 +1,232 @@
+  { rw [←one_mul (↑h.unit⁻¹ : A), ←hx.unit.inv_mul, mul_assoc, is_unit.unit_spec, mul_assoc,
+      h.mul_coe_inv, mul_one],
+    exact mul_mem this (star_mem hxS) },
+  refine le_of_is_closed_of_mem ℂ hS (mul_mem (star_mem hxS) hxS) _,
+  haveI := (is_self_adjoint.star_mul_self x).is_star_normal,
+  have hx' := elemental_star_algebra.is_unit_of_is_unit_of_is_star_normal hx,
+  convert (↑hx'.unit⁻¹ : elemental_star_algebra ℂ (star x * x)).prop using 1,
+  exact left_inv_eq_right_inv hx.unit.inv_mul (congr_arg coe hx'.unit.mul_inv),
+end
+
+/-- For a unital C⋆-subalgebra `S` of `A` and `x : S`, if `↑x : A` is invertible in `A`, then
+`x` is invertible in `S`. -/
+lemma star_subalgebra.coe_is_unit {S : star_subalgebra ℂ A} (hS : is_closed (S : set A)) {x : S} :
+  is_unit (x : A) ↔ is_unit x :=
+begin
+  refine ⟨λ hx, ⟨⟨x, ⟨(↑hx.unit⁻¹ : A), star_subalgebra.is_unit_coe_inv_mem hS hx x.prop⟩, _, _⟩,
+    rfl⟩, λ hx, hx.map S.subtype⟩,
+  exacts [subtype.coe_injective hx.mul_coe_inv, subtype.coe_injective hx.coe_inv_mul],
+end
+
+/-- **Spectral permanence.** The spectrum of an element is invariant of the `star_subalgebra` in
+which it is contained. -/
+lemma star_subalgebra.spectrum_eq {S : star_subalgebra ℂ A} (hS : is_closed (S : set A)) (x : S) :
+  spectrum ℂ x = spectrum ℂ (x : A) :=
+set.ext $ λ _, not_iff_not.2 (star_subalgebra.coe_is_unit hS).symm
+
+variables (a)
+
+/-- The natural map from `character_space ℂ (elemental_star_algebra ℂ a)` to `spectrum ℂ a` given
+by evaluating `φ` at `a`. This is essentially just evaluation of the `gelfand_transform` of `a`,
+but because we want something in `spectrum ℂ a`, as opposed to
+`spectrum ℂ ⟨a, elemental_star_algebra.self_mem ℂ a⟩` there is slightly more work to do. -/
+@[simps]
+noncomputable def elemental_star_algebra.character_space_to_spectrum (x : A) :
+  C(character_space ℂ (elemental_star_algebra ℂ x), spectrum ℂ x) :=
+{ to_fun := λ φ,
+  { val := φ ⟨x, self_mem ℂ x⟩,
+    property := by simpa only [star_subalgebra.spectrum_eq (elemental_star_algebra.is_closed ℂ x)
+      ⟨x, self_mem ℂ x⟩] using alg_hom.apply_mem_spectrum φ (⟨x, self_mem ℂ x⟩) },
+  continuous_to_fun := continuous_induced_rng.2 (map_continuous $
+    gelfand_transform ℂ (elemental_star_algebra ℂ x) ⟨x, self_mem ℂ x⟩) }
+
+lemma elemental_star_algebra.character_space_to_spectrum_bijective :
+  function.bijective (elemental_star_algebra.character_space_to_spectrum a) :=
+begin
+  refine ⟨λ φ ψ h, star_alg_hom_class_ext ℂ (map_continuous φ) (map_continuous ψ)
+    (by simpa only [elemental_star_algebra.character_space_to_spectrum, subtype.mk_eq_mk,
+      continuous_map.coe_mk] using h), _⟩,
+  rintros ⟨z, hz⟩,
+  set a' : elemental_star_algebra ℂ a := ⟨a, self_mem ℂ a⟩,
+  rw [(show a = a', from rfl), ←star_subalgebra.spectrum_eq
+    (elemental_star_algebra.is_closed ℂ a) a', ←spectrum.gelfand_transform_eq a',
+    continuous_map.spectrum_eq_range] at hz,
+  obtain ⟨φ, rfl⟩ := hz,
+  exact ⟨φ, rfl⟩,
```

Comment:
Could you extract this part of the proof to a lemma? I think it could be useful by itself.

Reply:
Sorry, which part of the proof do you mean exactly? I'm not sure what lemma you want. Whatever it is, I somewhat doubt that we want it (since it will be a result involving the spectrum of `a` relative to its `elemental_star_subalgebra`, and that's only of interest in specialized situations).

Reply:
IIRC I just meant that we should have a lemma saying that any element of the spectrum of `a` is the image of `a` under a character, instead of going back to the Gelfand transform and using `continuous_map.spectrum_eq_range`. So yes, I did very bad at explaining which "part of the proof" I meant, sorry for that :sweat_smile: